src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -580,28 +570,22 @@ string Visitor::visit(Parser::PnameNsContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-SparqlQleverVisitor::PrefixMap SparqlQleverVisitor::visit(
-    SparqlAutomaticParser::PrologueContext* ctx) {
+void Visitor::visit(Parser::PrologueContext* ctx) {
   if (!ctx->baseDecl().empty()) {
-    reportError(ctx->baseDecl(0), "BaseDecl is not supported.");
+    reportNotSupported(ctx->baseDecl(0), "BASE declarations are");
   }
   for (const auto& prefix : ctx->prefixDecl()) {
     visit(prefix);
   }
-  // TODO: we return a part of our internal state here. This will go away when
-  //  queries can be parsed completely with ANTLR.
-  return _prefixMap;
 }
 
 // ____________________________________________________________________________________
-SparqlPrefix SparqlQleverVisitor::visit(
-    SparqlAutomaticParser::BaseDeclContext* ctx) {
-  reportError(ctx, "BaseDecl is not supported.");
+SparqlPrefix Visitor::visit(Parser::BaseDeclContext* ctx) {
```

Comment:
Should probably return `void` (if we ever support them, they will also update the internal state).

Reply:
You are right.